### PR TITLE
Fix missing parameter initialization in Axes.specgram()

### LINKED ISSUE #9100: Parameter Fs initialization missing in Axes.specgram()
### Bug report

**Bug summary**

the docstring of parameter `Fs` of `matplotlib.pyplot.specgram()` says that the default value is `2` but the default value in the function signature is `None`.

**Code for reproduction**

point your browser to https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.specgram.html or do:

```python
import matplotlib.pyplot as plt
help(plt.specgram)
```

**Actual outcome**

```
specgram(x, NFFT=None, Fs=None, Fc=None, ...
...
Fs : scalar
        The sampling frequency (samples per time unit).  It is used
        to calculate the Fourier frequencies, freqs, in cycles per time
        unit. The default value is 2.
```

**Expected outcome**

Either write `None` in the docstring or make change the default value of Fs to 2.

**Matplotlib version**

  * Operating System: OSX 10.11.6
  * Matplotlib Version: 2.0.2
  * Python Version: 3.5.4
  * Jupyter Version (if applicable): notebook server 5.0.0, IPython 6.1.0
  * Other Libraries: N/A

Installed with pip in a conda environment.

### PR DESCRIPTION
**Pull request by Stefan-Mitic**
## PR Summary
Added check to set Fs in _axes.specgram() to be Fs = freq[-1] * 2.
Added a test to test_axes.py to check if Fs is None that it won't throw an error.

Closes #9100

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -4166,6 +4166,17 @@ def test_specgram_angle():
                                 scale="dB")
 
 
+def test_specgram_fs_none():
+    """ Test axes.specgram when Fs is None, should not throw error """
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: The style checking is unhappy about the extra spaces around the docstring.

**Comment by tacaswell** [comment]
Thanks @Stefan-Mitic !  Other than the issue with the style, this looks good to me!

File: lib/matplotlib/tests/test_axes.py (hunk 1 above)
Comment (Stefan-Mitic) [reply lib/matplotlib/tests/test_axes.py]: Thanks! Fixed.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -7649,6 +7649,8 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
         Z = np.flipud(Z)
 
         if xextent is None:
+            if Fs is None:
```
Comment (timhoffm) [inline lib/matplotlib/axes/_axes.py]: I'd move this up to where we redefine the other constants like `NFFT` and `Fc`.

Comment (Stefan-Mitic) [reply lib/matplotlib/axes/_axes.py]: Sure! Thanks for the suggestion.

File: lib/matplotlib/tests/test_axes.py [hunk 3]
```diff
@@ -4166,6 +4166,18 @@ def test_specgram_angle():
                                 scale="dB")
 
 
+def test_specgram_fs_none():
+    """Test axes.specgram when Fs is None, should not throw error."""
+
+    try:
+        spec, freqs, t, im = plt.specgram(np.ones(300), Fs=None)
+    except Exception as e:
+        raise pytest.fail("DID RAISE {0}".format(e))
```
Comment (timhoffm) [inline lib/matplotlib/tests/test_axes.py:4175]: You don't need the `try ... except` pytest will fail if there is an Execption; and the message does not contain more information than the exception, which will be printed anyway.

Comment (Stefan-Mitic) [reply lib/matplotlib/tests/test_axes.py:4175]: Hmm alright, so I'll just run ```plt.specgram``` without the ```try...except``` if pytest handles it.

**Comment by timhoffm** [comment]
Thanks, and congratulations on your first contribution to Matplotlib. We hope to see you back some time.

**Pull request merged**